src/import/FSHImporter.ts

```diff
@@ -1916,6 +1936,75 @@ export class FSHImporter extends FSHVisitor {
     }
   }
 
+  /**
+   * For each rule, expand the path based on context so that the path is complete
+   * @param docs - The array of FSHDocuments with entities containing rules
+   */
+  private expandContextPaths(docs: FSHDocument[]): void {
+    docs.forEach(doc => {
+      [
+        ...doc.codeSystems.values(),
+        ...doc.extensions.values(),
+        ...doc.profiles.values(),
+        ...doc.valueSets.values(),
+        ...doc.instances.values(),
+        ...doc.appliedRuleSets.values(),
+        ...doc.ruleSets.values()
+      ].forEach(definition => {
+        let context: string[] = [];
+        // NOTE: indentWidth is hard coded for now, but if we later allow non-multiples of 2 to indent, we can change that
+        const indentWidth = 2;
+        const baseIndent = definition.rules[0]?.sourceInfo.location.startColumn;
+        definition.rules.forEach((rule: Rule) => {
+          // We require that all rules are indented a multiple of indentWidth
+          const ruleIndent = rule.sourceInfo.location.startColumn - baseIndent;
+          if (ruleIndent % indentWidth !== 0 || ruleIndent < 0) {
+            logger.error(
+              `Unable to determine context for rule indented ${ruleIndent} space(s). Rules must be indented in multiples of ${indentWidth} space(s)`,
+              rule.sourceInfo
+            );
+            return;
+          }
+
+          // And we require that rules are not indented too deeply
+          const contextIndex = ruleIndent / indentWidth;
+          if (contextIndex > context.length) {
+            logger.error(
+              `Cannot determine context of rule since it is indented too deeply. Rules must be indented in increments of ${indentWidth} space(s)`,
+              rule.sourceInfo
+            );
+            return;
+          }
+
+          // If the element is not indented, just reset the context
+          if (contextIndex === 0) {
+            context = [rule.path];
+            return;
+          }
+
+          // Insert Rules cannot be given context, since they do not have a "path".
+          // Rules on VS and CS cannot be indented, for style consistency we are still checking them here
+          if (
+            rule instanceof InsertRule ||
+            rule instanceof ConceptRule ||
+            rule instanceof ValueSetComponentRule
+          ) {
+            logger.error(
+              `Rule of type ${rule.constructorName} cannot be indented to indicate context. The rule will be processed as if it is not indented`,
+              rule.sourceInfo
+            );
+            return;
+          }
+
+          // Replace '[+]' with '[=]' in higher level ccontexts, since children are at the same index as the parent contexts
```

Comment:
I'm sorry I ccan't unsee it now.

```suggestion
          // Replace '[+]' with '[=]' in higher level contexts, since children are at the same index as the parent contexts
```